### PR TITLE
Image picker

### PR DESCRIPTION
**Pull request by simplekjl**
Image picker with chooser intent action

### CODE REVIEW

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 1]
```diff
@@ -48,7 +78,10 @@ class MainActivity : AppCompatActivity() {
         //Image Picker button
 
         viewBinding.photoPickerButton.setOnClickListener {
-            // TODO do something
+            var intent: Intent = Intent(Intent.ACTION_GET_CONTENT)
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:81]:
```
var intent = Intent(Intent.ACTION_GET_CONTENT).apply {
    type = "image/jpeg"
    putExtra(Intent.EXTRA_LOCAL_ONLY, true)
}
```

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 2]
```diff
@@ -19,13 +34,23 @@ class MainActivity : AppCompatActivity() {
 
     val ANONYMOUS = "anonymous"
     val DEFAULT_MSG_LENGTH_LIMIT = 1000
+    val RC_SIGN_IN: Int = 12
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:37]: These can be private if not being used in other classes.

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 3]
```diff
@@ -73,36 +106,155 @@ class MainActivity : AppCompatActivity() {
         //Send messages action
         viewBinding.sendButton.setOnClickListener { _ ->
 
-            var message = CustomMessage(viewBinding.messageEditText.text.toString(),userName,null)
+            var message = CustomMessage(viewBinding.messageEditText.text.toString(), userName, null)
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:109]: `val` would work just fine.

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 4]
```diff
@@ -73,36 +106,155 @@ class MainActivity : AppCompatActivity() {
...
+                override fun onChildMoved(p0: DataSnapshot, p1: String?) {
+
+                }
+
+                override fun onChildChanged(p0: DataSnapshot, p1: String?) {
+
+                }
+
+                override fun onChildAdded(p0: DataSnapshot, p1: String?) {
+                    var customMessage = p0.getValue(CustomMessage::class.java)
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:163]: `val` would work just fine.

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 5]
```diff
@@ -73,36 +106,155 @@ class MainActivity : AppCompatActivity() {
...
+                    mMessagesAdapter.add(customMessage)
+                }
 
-            override fun onChildMoved(p0: DataSnapshot, p1: String?) {
+                override fun onChildRemoved(p0: DataSnapshot) {
 
+                }
             }
+            // adding the listener to the reference
+            mMessagesDatabaseReference.addChildEventListener(mChildEventListener!!)
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:173]: Avoid `!!`

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 6]
```diff
@@ -73,36 +106,155 @@ class MainActivity : AppCompatActivity() {
...
+        }
+    }
+
+    private fun detachDatabaseListener() {
+        if (mChildEventListener != null) {
+            mMessagesDatabaseReference.removeEventListener(mChildEventListener!!)
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:179]: Avoid `!!`

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt [hunk 7]
```diff
@@ -73,36 +106,155 @@ class MainActivity : AppCompatActivity() {
...
+        if (resultCode == RC_SIGN_IN) {
+            if (resultCode == Activity.RESULT_OK) {
+                Toast.makeText(this, "Signed in!", Toast.LENGTH_SHORT).show()
+
+            } else if (resultCode == Activity.RESULT_CANCELED) {
+                Toast.makeText(this, "Signed out!", Toast.LENGTH_SHORT).show()
+                finish()
             }
+        } else if (requestCode == RC_PHOTO_PICKER && resultCode == Activity.RESULT_OK) {
+            var selectedImageUri: Uri = data?.data ?: Uri.EMPTY
```
Comment (curioussabhi) [inline app/src/main/java/com/simplekjl/letschat/MainActivity.kt:210]: `val` ?

**Review by curioussabhi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/simplekjl/letschat/MainActivity.kt (hunk 2 above)
Comment (simplekjl) [reply app/src/main/java/com/simplekjl/letschat/MainActivity.kt:37]: I fixed them on the next branch, adding them into the companion object :D

**Pull request merged**